pkg/cmd/stack/destroy.go

```diff
@@ -44,17 +127,88 @@ func Destroy(ctx context.Context, s *model.Stack) error {
 		return fmt.Errorf("error initializing stack client: %s", err)
 	}
 
-	exists, err := helm.ReleaseExist(action.NewList(actionConfig), s.Name)
+	exists, err := helmReleaseExist(action.NewList(actionConfig), s.Name)
 	if err != nil {
 		return fmt.Errorf("error listing stacks: %s", err)
 	}
-	if !exists {
-		return fmt.Errorf("stack %s does not exist", s.Name)
+	if exists {
+		uClient := action.NewUninstall(actionConfig)
+		if _, err := uClient.Run(s.Name); err != nil {
+			return fmt.Errorf("error destroying stack '%s': %s", s.Name, err)
+		}
+	}
+	return nil
+}
+
+func destroyServicesNotInStack(ctx context.Context, s *model.Stack, c *kubernetes.Clientset) error {
+	dList, err := deployments.List(ctx, s.Namespace, s.GetLabelSelector(), c)
+	if err != nil {
+		return err
+	}
+	for _, d := range dList {
+		if _, ok := s.Services[d.Name]; !ok {
+			if err := deployments.Destroy(ctx, d.Name, d.Namespace, c); err != nil {
+				return err
+			}
+		}
+	}
+
+	sfsList, err := statefulsets.List(ctx, s.Namespace, s.GetLabelSelector(), c)
+	if err != nil {
+		return err
+	}
+	for _, sfs := range sfsList {
+		if _, ok := s.Services[sfs.Name]; !ok {
+			if err := statefulsets.Destroy(ctx, sfs.Name, sfs.Namespace, c); err != nil {
+				return err
+			}
+		}
 	}
 
-	uClient := action.NewUninstall(actionConfig)
-	if _, err := uClient.Run(s.Name); err != nil {
-		return fmt.Errorf("error destroying stack '%s': %s", s.Name, err)
+	svcList, err := services.List(ctx, s.Namespace, s.GetLabelSelector(), c)
+	if err != nil {
+		return err
+	}
+	for _, svc := range svcList {
+		if _, ok := s.Services[svc.Name]; !ok {
+			if err := services.Destroy(ctx, svc.Name, svc.Namespace, c); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func waitForPodsToBeDestroyed(ctx context.Context, s *model.Stack, c *kubernetes.Clientset) error {
+	ticker := time.NewTicker(100 * time.Millisecond)
+	timeout := time.Now().Add(300 * time.Second)
```

Comment:
can we have the timeouts on a single location? (I'm sure at some point we'll want to make them parametric.